packages/components/src/composite/types.ts

```diff
@@ -3,16 +3,14 @@
  */
 import type * as Ariakit from '@ariakit/react';
 
-export type CompositeContextProps =
-	| {
-			/**
-			 * Object returned by the `useCompositeStore` hook.
-			 */
-			store: Ariakit.CompositeStore;
-	  }
-	| undefined;
+export type CompositeContextProps = {
+	/**
+	 * The component store, used for advanced usage of the component.
+	 */
+	store?: unknown;
```

Comment:
Sorry if this was already discussed somewhere — why is this `unknown` now? Might need a code comment since it was non-obvious to me.

Reply:
It was discussed in one of the (many) inline comments, so quite easy to miss 😅 

We thought that, in order to discourage `store` usage, we'd change the type to `unknown` — which still allows TypeScript to correctly validate scenarios like context forwarding (check the "With Slot Fill" example), but it wouldn't give insight into what `store` is and does.

What do you think? Too much?

Happy to make changes in a follow-up.